webapp/src/main/java/life/qbic/views/login/passwordreset/PasswordResetHandler.java

```diff
@@ -3,55 +3,93 @@
+    this.passwordReset = passwordReset;
+  }
 
-    @Override
-    public void handle(ResetPasswordLayout layout) {
-        if (registeredPasswordResetLayout != layout) {
-            this.registeredPasswordResetLayout = layout;
-            addClickListeners();
-        }
+  @Override
+  public void handle(ResetPasswordLayout layout) {
+    if (registeredPasswordResetLayout != layout) {
+      this.registeredPasswordResetLayout = layout;
+      addClickListeners();
     }
+  }
 
-    private void addClickListeners() {
-        registeredPasswordResetLayout.sendButton.addClickListener(buttonClickEvent ->
-            passwordReset.resetPassword(registeredPasswordResetLayout.email.getValue()));
-        registeredPasswordResetLayout.sendButton.addClickShortcut(Key.ENTER);
+  private void addClickListeners() {
+    registeredPasswordResetLayout.sendButton.addClickListener(
+        buttonClickEvent -> {
+          clearNotifications();
+          passwordReset.resetPassword(registeredPasswordResetLayout.email.getValue());
+        });
+    registeredPasswordResetLayout.sendButton.addClickShortcut(Key.ENTER);
 
-        registeredPasswordResetLayout.linkSentLayout.loginButton.addClickListener(buttonClickEvent ->
-                registeredPasswordResetLayout.linkSentLayout.getUI().ifPresent(ui -> ui.navigate("login")));
-    }
+    registeredPasswordResetLayout.linkSentLayout.loginButton.addClickListener(
+        buttonClickEvent ->
+            registeredPasswordResetLayout
+                .linkSentLayout
+                .getUI()
+                .ifPresent(ui -> ui.navigate("login")));
+  }
 
-    @Override
-    public void onPasswordResetSucceeded() {
-        registeredPasswordResetLayout.linkSentLayout.setVisible(true);
-        registeredPasswordResetLayout.enterEmailLayout.setVisible(false);
-    }
+  public void clearNotifications() {
+    registeredPasswordResetLayout.enterEmailLayout.removeNotifications();
+  }
+
+  public void showError(String title, String description) {
+    clearNotifications();
+    ErrorMessage errorMessage = new ErrorMessage(title, description);
+    registeredPasswordResetLayout.enterEmailLayout.setNotification(errorMessage);
+  }
+
+  private void showPasswordResetFailedError(String error, String description) {
+    showError(error, description);
+  }
+
+  @Override
+  public void onPasswordResetSucceeded() {
+    registeredPasswordResetLayout.linkSentLayout.setVisible(true);
+    registeredPasswordResetLayout.enterEmailLayout.setVisible(false);
+  }
 
-    @Override
-    public void onPasswordResetFailed() {
-        //todo
-        throw new NotImplementedException();
+  @Override
+  public void onPasswordResetFailed(ApplicationResponse response) {
+    for (RuntimeException failure : response.failures()) {
+      if (failure instanceof EmailAddress.EmailValidationException) {
+        showPasswordResetFailedError(failure.getMessage(), "Please provide a valid email address.");
```

Comment:
I think we should not use the exception message here, and remain in control what we display to the user. Based on the exception type, we can define that easily.

Reply:
I am ok with that. Will do